Modules/xSharePoint/DSCResources/MSFT_xSPFarmSolution/MSFT_xSPFarmSolution.psm1

```diff
@@ -0,0 +1,341 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]  [String]   $Name,
+        [parameter(Mandatory = $true)]  [String]   $LiteralPath,
+        [parameter(Mandatory = $false)] [String[]] $WebApplications = @(),
+        [parameter(Mandatory = $false)] [ValidateSet("Present","Absent")] 
+                                        [String]   $Ensure = "Present",
+        [parameter(Mandatory = $false)] [String]   $Version = "1.0.0.0",
+        [parameter(Mandatory = $false)] [Boolean]  $Deployed = $true,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    Write-Verbose -Message "Getting farm solution '$Name'..."
+
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+
+        $solution = Get-SPSolution -Identity $params.Name -ErrorAction SilentlyContinue -Verbose:$false
+
+        if ($null -ne $solution) { 
+            $currentState = "Present" 
+            $deployed = $solution.Deployed
+            $version = $Solution.Properties["Version"]
+            $deployedWebApplications = @($solution.DeployedWebApplications | select -ExpandProperty Url)
+            $ContainsGlobalAssembly = $solution.ContainsGlobalAssembly
+        } else { 
+            $currentState = "Absent" 
+            $deployed = $false
+            $version = "0.0.0.0"
+            $deployedWebApplications = @()
+            $ContainsGlobalAssembly = $false
+        }
+
+        return @{
+            Name            = $params.Name
+            LiteralPath     = $LiteralPath
+            Deployed        = $deployed
+            Ensure          = $currentState
+            Version         = $version
+            WebApplications = $deployedWebApplications
+            ContainsGlobalAssembly = $ContainsGlobalAssembly
+        }
+    }
+
+    return $result
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
```

Comment:
Can you remove the output type here - we expect no output from the set method normally